pkg/cmd/samples/create.go

```diff
@@ -181,3 +188,97 @@ To see supported samples, run 'stripe samples list'`, args[0])
 
 	return nil
 }
+
+// PromptSampleConfig prompts the user to select the integration they want to use
+// (if available) and the language they want the integration to be.
+func PromptSampleConfig(sample samples.Samples) (*samples.SelectedConfig, error) {
+	var selectedConfig samples.SelectedConfig
+
+	if sample.SampleConfig.HasIntegrations() {
+		integration, err := integrationSelectPrompt(&sample.SampleConfig)
+		if err != nil {
+			return nil, err
+		}
+		selectedConfig.Integration = integration
+	} else {
+		selectedConfig.Integration = &sample.SampleConfig.Integrations[0]
+	}
+
+	if selectedConfig.Integration.HasMultipleClients() {
+		client, err := clientSelectPrompt(selectedConfig.Integration.Clients)
+		if err != nil {
+			return nil, err
+		}
+		selectedConfig.Client = client
+	} else {
+		selectedConfig.Client = ""
+	}
+
+	if selectedConfig.Integration.HasMultipleServers() {
+		server, err := serverSelectPrompt(selectedConfig.Integration.Servers)
+		if err != nil {
+			return nil, err
+		}
+		selectedConfig.Server = server
+	} else {
+		selectedConfig.Server = ""
+	}
+
+	return &selectedConfig, nil
+}
+
+func selectOptions(template, label string, options []string) (string, error) {
+	color := ansi.Color(os.Stdout)
+
+	templates := &promptui.SelectTemplates{
+		Selected: color.Green("✔").String() + ansi.Faint(fmt.Sprintf(" Selected %s: {{ . | bold }} ", template)),
+	}
+	prompt := promptui.Select{
+		Label:     label,
+		Items:     options,
+		Templates: templates,
+	}
+
+	_, result, err := prompt.Run()
+
+	if err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+func clientSelectPrompt(clients []string) (string, error) {
+	selected, err := selectOptions("client", "Which client would you like to use", clients)
+	if err != nil {
+		return "", err
+	}
+
+	return selected, nil
+}
+
+func integrationSelectPrompt(sc *samples.SampleConfig) (*samples.SampleConfigIntegration, error) {
+	selected, err := selectOptions("integration", "What type of integration would you like to use", sc.IntegrationNames())
+	if err != nil {
+		return nil, err
+	}
+
+	var selectedIntegration *samples.SampleConfigIntegration
+
+	for i, integration := range sc.Integrations {
+		if integration.Name == selected {
+			selectedIntegration = &sc.Integrations[i]
+		}
+	}
+
+	return selectedIntegration, nil
+}
+
+func serverSelectPrompt(servers []string) (string, error) {
+	selected, err := selectOptions("server", "What server would you like to use", servers)
+	if err != nil {
+		return "", err
+	}
+
+	return selected, nil
```

Comment:
all those are fine here indeed. all the Prompts, Options, etc. (user input) should live here. The Cobra cmd should only have the responsibility to take in user input, and then pass it to the samples package as arguments to a "create" func.